packages/babel-helpers/src/helpers.js

```diff
@@ -461,6 +461,8 @@ helpers.set = template(`
 
       if (parent !== null) {
         set(parent, property, value, receiver);
+      } else {
+        receiver[property] = value;
       }
     } else if ("value" in desc && desc.writable) {
       desc.value = value;
```

Comment:
The helper also just silently does nothing if not writable (needs to throw in strict mode), but solving that is a harder problem.